magda-typescript-common/src/authorization-api/ApiClient.ts

```diff
@@ -2,56 +2,81 @@ require("isomorphic-fetch");
 
 import { User } from "./model";
 import { Maybe } from "tsmonad";
+import buildJwt from "../session/buildJwt";
 
 export default class ApiClient {
-    constructor(readonly baseUrl: string) {}
+    
+    private jwt: string = null;
+    private requestInitOption: RequestInit = null;
 
-    getUser(userId: string): Promise<Maybe<User>> {
-        return this.handleGetResult(
-            fetch(`${this.baseUrl}/private/users/${userId}`)
+    constructor(readonly baseUrl: string, jwtSecret: string = null, userId: string = null) {
+        if(jwtSecret && userId) {
+            this.jwt = buildJwt(jwtSecret,userId);
+        }
+        this.requestInitOption={
+            headers: {
+                "X-Magda-Session" : this.jwt
+            }
+        }
+    }
+
+    getMergeRequestInitOption(extraOptions: RequestInit = null): RequestInit{
+        let options: RequestInit = this.requestInitOption ? {...this.requestInitOption} : {};
```

Comment:
Could you use [lodash merge](https://lodash.com/docs/4.17.4#merge) for this?

Reply:
No problem. Will use lodash for this one.

Reply:
done in commit b21a0874c3ec4037e46f3b24dab6931a83f1979e